orttraining/orttraining/training_ops/cuda/communication/nccl_service.h

```diff
@@ -0,0 +1,185 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#ifdef USE_NCCL
+
+#pragma once
+#include <condition_variable>
+#include <list>
+#include <mutex>
+#include <map>
+#include <vector>
+#include <iostream>
+#include <string>
+#include <thread>
+
+#include <nccl.h>
+#include <mpi.h>
+
+namespace onnxruntime {
+namespace cuda {
+
+struct NcclTask final {
+  // Attributes for communication operator.
+  enum class Type { SEND,
+                    RECV,
+                    ALLREDUCE };
+  // Operator to perform.
+  Type type;
+  // For Send, this field is destination device's ID.
+  // For Recv, this field is source device's ID.
+  std::vector<int> peers;
+
+  // Attributes for memory location.
+  // GPU memory pointer.
+  void* ptr;
+  // Number of bytes to send/recv.
+  size_t size;
+
+  // Scheduler flag.
+  bool is_enqueued;
+  bool is_finished;
+
+  // Debug information
+  std::string info;
```

Comment:
Please append '_' to variable names to indicate they are struct fields.

Reply:
Same for `NcclTaskGroup` class.

---
In reply to: [485959762](https://github.com/microsoft/onnxruntime/pull/5054#discussion_r485959762) [](ancestors = 485959762)